### PR TITLE
fix(device-client) - Fixed xml comment warnings in device/src

### PR DESCRIPTION
**Pull request by vinagesh**
## Checklist
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `master` branch.

### CODE REVIEW

File: iothub/device/src/MethodResponseStatusCode.cs [hunk 1]
```diff
@@ -8,8 +8,11 @@ namespace Microsoft.Azure.Devices.Client
     /// </summary>
     public enum MethodResponseStatusCode
     {
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
```
Comment (vinagesh) [inline iothub/device/src/MethodResponseStatusCode.cs:11]: Adding comments here seemed redundant. If someone has a better idea, let me know.

File: iothub/device/src/MethodRequest.cs [hunk 2]
```diff
@@ -6,20 +6,46 @@
 
 namespace Microsoft.Azure.Devices.Client
 {
+    /// <summary>
+    /// The data structure that represents a method request.
+    /// </summary>
     public sealed class MethodRequest
     {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MethodRequest"/> class with without any method data and extra time for
```
Comment (vinagesh) [inline iothub/device/src/MethodRequest.cs:15]: This is what I found by looking in code. Someone with more context please validate.

File: iothub/device/src/DeviceAuthenticationWithTpm.cs [hunk 3]
```diff
@@ -18,13 +18,26 @@ public sealed class DeviceAuthenticationWithTpm : DeviceAuthenticationWithTokenR
     {
         private SecurityProviderTpm _securityProvider;
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DeviceAuthenticationWithTpm"/> class with default
+        /// time to live of 1 minute and default buffer percentage value of 15.
```
Comment (abhipsaMisra) [inline iothub/device/src/DeviceAuthenticationWithTpm.cs:23]: isn't the default ttl 1 hour?

File: iothub/device/src/Http1TransportSettings.cs [hunk 4]
```diff
@@ -15,20 +15,36 @@ public sealed class Http1TransportSettings : ITransportSettings
     {
         private static readonly TimeSpan s_defaultOperationTimeout = TimeSpan.FromSeconds(60);
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Http1TransportSettings"/> class with default proxy settings.
```
Comment (abhipsaMisra) [inline iothub/device/src/Http1TransportSettings.cs:19]: I wouldn't specify that we initialize it with any default proxy setting

File: iothub/device/src/DeviceAuthenticationWithTpm.cs [hunk 5]
```diff
@@ -18,13 +18,26 @@ public sealed class DeviceAuthenticationWithTpm : DeviceAuthenticationWithTokenR
...
             _securityProvider = securityProvider ?? throw new ArgumentNullException(nameof(securityProvider));
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DeviceAuthenticationWithTpm"/> class.
+        /// </summary>
+        /// <param name="deviceId">Device Identifier.</param>
+        /// <param name="securityProvider">Device Security Provider settings for TPM Hardware Security Modules.</param>
+        /// <param name="suggestedTimeToLiveSeconds">Token time to live suggested value.</param>
+        /// <param name="timeBufferPercentage">Time buffer before expiry when the token should be renewed expressed as percentage of the time to live.</param>
```
Comment (timtay-microsoft) [inline iothub/device/src/DeviceAuthenticationWithTpm.cs:40]: The wording here could use an example to really clarify that if they want a SAS token to live for 85% of life before proactive renewal, then this value should be 15.

File: iothub/device/src/Http1TransportSettings.cs (hunk 4 above)
Comment (timtay-microsoft) [inline iothub/device/src/Http1TransportSettings.cs:19]: Does "Default proxy settings" mean that no proxy will be used?

File: iothub/device/src/Http1TransportSettings.cs [hunk 6]
```diff
@@ -15,20 +15,36 @@ public sealed class Http1TransportSettings : ITransportSettings
...
+        /// </summary>
         public Http1TransportSettings()
         {
             Proxy = DefaultWebProxySettings.Instance;
         }
 
+        /// <summary>
+        /// Gets the transport type for this settings class.
+        /// </summary>
+        /// <returns>HyperText Transfer Protocol transport type.</returns>
```
Comment (timtay-microsoft) [inline iothub/device/src/Http1TransportSettings.cs:29]: Maybe link the enum value directly here instead?

File: iothub/device/src/MethodRequest.cs [hunk 7]
```diff
@@ -6,20 +6,46 @@
...
         {
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MethodRequest"/> class.
+        /// </summary>
+        /// <param name="name">The method name.</param>
+        /// <param name="data">The method data.</param>
+        /// <param name="responseTimeout">The method timeout value.</param>
+        /// <param name="connectionTimeout">The device connection timeout value.</param>
```
Comment (timtay-microsoft) [inline iothub/device/src/MethodRequest.cs:48]: It's worth spelling out what these response/connect timeouts actually do. You can read about it in the paragraph in [this doc](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-direct-methods#method-invocation)

File: iothub/device/src/MethodRequest.cs [hunk 8]
```diff
@@ -28,18 +54,33 @@ public MethodRequest(string name, byte[] data, TimeSpan? responseTimeout, TimeSp
             ConnectionTimeout = connectionTimeout;
         }
 
+        /// <summary>
+        /// The method name.
+        /// </summary>
         public string Name { get; private set; }
 
+        /// <summary>
+        /// The method data.
```
Comment (timtay-microsoft) [inline iothub/device/src/MethodRequest.cs:63]: It's worth noting that this data must be serialized json

**Review by timtay-microsoft — APPROVED** [review]
Just a few small comments, otherwise looks good

File: iothub/device/src/MethodRequest.cs (hunk 7 above)
Comment (timtay-microsoft) [reply iothub/device/src/MethodRequest.cs:48]: Also worth adding that there are service side limits to these values, also found in the above doc

File: iothub/device/src/Http1TransportSettings.cs (hunk 4 above)
Comment (abhipsaMisra) [reply iothub/device/src/Http1TransportSettings.cs:19]: default proxy behavior - whatever is specified at your OS level

File: iothub/device/src/MethodRequest.cs [hunk 9]
```diff
@@ -28,18 +54,37 @@ public MethodRequest(string name, byte[] data, TimeSpan? responseTimeout, TimeSp
...
         [System.Diagnostics.CodeAnalysis.SuppressMessage(
             "Performance",
             "CA1819:Properties should not return arrays",
             Justification = "Cannot change property types on public classes.")]
         public byte[] Data { get; private set; }
 
+        /// <summary>
+        /// The method response timeout value. This value is the amount of time that IoT Hub service must await for
+        /// completion of a direct method execution on a device. The minimum and maximum values are 5 and 300 seconds.
+        /// </summary>
```
Comment (abhipsaMisra) [inline iothub/device/src/MethodRequest.cs:74]: it is worth specifying here that these values for `MethodRequest` are relevant only when invoking methods from one edge module to another. For device client operations the user will receive the `MethodRequest` as a part of the direct method callback invoked, so they wouldn't really be setting these timeout values.

File: iothub/device/src/MethodResponseStatusCode.cs (hunk 1 above)
Comment (abhipsaMisra) [reply iothub/device/src/MethodResponseStatusCode.cs:11]: The description for 500 = UserCodeException is a bit confusing to me; isn't 500 always a service error?

for the warning - how about we add it to the suppressions file instead, might look cleaner. It looks like we have one for device client: https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/src/GlobalSuppressions.cs

File: iothub/device/src/MethodRequest.cs (hunk 2 above)
Comment (abhipsaMisra) [reply iothub/device/src/MethodRequest.cs]: the default value for timeouts here should be null -> not specified -> 30secs for response timeout and 0s for connect: https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-direct-methods#method-invocation

**Review by abhipsaMisra — APPROVED** [review]
just the one comment about method request timeout value, everything else looks good 👍🏽

File: iothub/device/src/MethodResponseStatusCode.cs (hunk 1 above)
Comment (azabbasi) [reply iothub/device/src/MethodResponseStatusCode.cs:11]: What if we mention that these values are interchangable with actual http status codes?
Anyways, if you decide not to add comments for them, you can supress them in a single attribute on the class

**Review by azabbasi — APPROVED** [review]
Just a small comment on the MethodResponseStatusCode file.

**Pull request merged**